packages/wonder-blocks-core/components/id-provider.js

```diff
@@ -30,24 +31,25 @@ type Props = {|
 |};
 
 /**
- * This is a Dialog wrapper that returns a unique Modal Title for A11Y purposes.
- * This way, the wrapped Dialog will receive this custom ID and will use it to refer to a
- * visible dialog title.
+ * This is a wrapper that returns a unique Identifier for A11Y purposes.
```

Comment:
Now that this is public it might be good to also explain when this should be used contrasted with UniqueIDProvider.

Reply:
Agreed. I think it is likely that IDProvider is the preferred API in most cases, with UniqueIDProvider being a "break the glass" option when implementing more complex scenarios.

Reply:
Also, I don't think `for A11Y purposes` is necessary here. There are many reasons one may need this behavior; a11y is just one of them.